src/java.base/share/classes/jdk/internal/classfile/impl/StackMapGenerator.java

```diff
@@ -1058,23 +1055,26 @@ void setLocalsFromArg(String name, MethodTypeDesc methodDesc, boolean isStatic,
             }
             for (int i = 0; i < methodDesc.parameterCount(); i++) {
                 var desc = methodDesc.parameterType(i);
-                if (!desc.isPrimitive()) {
-                    setLocalRawInternal(localsSize++, Type.referenceType(desc));
-                } else switch (desc.descriptorString().charAt(0)) {
-                    case 'J' -> {
-                        setLocalRawInternal(localsSize++, Type.LONG_TYPE);
-                        setLocalRawInternal(localsSize++, Type.LONG2_TYPE);
+                if (desc == CD_void) throw new AssertionError("Should not reach here");
+                Type type;
+                if (desc instanceof PrimitiveClassDescImpl) {
```

Comment:
Can we use `desc.isPrimitive()`?

Reply:
I have tried isPrimitive(). The codeSize of the virtual function call isPrimitive() is 2 bytes larger than instanceof. The performance of instanceof should be better. But isPrimitive() is a public API and easier to understand. If you think isPrimitive is better, I think it's OK too.

Reply:
```java
ClassDesc {
    default boolean isPrimitive() {
        return descriptorString().length() == 1;
    }       
}
```

Using isPrimitive will result in three virtual function calls, isPrimitive/descriptorString/length. C2 can be optimized inline, but one more reference access to descriptorString is necessary. Considering the possible slowdown caused by cache miss, can we keep using instanceof?